draft-jmiller-json-web-proof.md

```diff
@@ -70,49 +71,25 @@ when, and only when, they appear in all capitals, as shown here.
+Algorithm definitions that support JWPs will be done in separate companion specifications - just as the [JSON Web Algorithms] [@RFC7518] specification does for JWS and JWE [@RFC7516].
 
-## Payload Headers
-
-With multiple payloads, there must also be multiple protected headers in order to safely identify what each payload is and contains.  These headers can themselves easily become a correlation factor if the signer is putting unique values in them or generating unique arragements of headers and their payloads.  In order to minimize these possibilities, JWP places payload headers in the JWK [@!RFC7517] definition so that they are common across all uses of that JWK.
-
-Privacy preserving algorithms have a common pattern of supporting a fixed number of "slots" (also called "messages" or "attributes").  Often they are fixed at the time the public key is created and sometimes they are also pre-defined for a use-case with a specific number of attributes.  This is done to minimize any correlatable signals, to prevent a verifier from categorizing based on if the slots are of variable lengths even when only a subset are revealed in the proof.
+# JWP Format
 
-The key used for a JWP that is identified either by its header or by context MUST have an associated ordered array of payload definitions.  If defined by a JWK, this specification registers a `payloads` parameter with an array value, containing one object for each payload with its associated header values.
+A JWP always contains a protected header along with one or more specific payloads.  The payloads are always serialized and processed as an ordered array.
 
-If the JWK includes a `use` parameter this specification registers a `prf` (proof) value that MUST be used in order for the JWK to be valid for a JWP.
+An individual payload may contain structured information such as a JSON document or be a simple value such as a number, string, or even a binary image.  Some algorithms may support payload values that are cryptographic values such as elliptic curve points or blinded secrets.
 
-For example:
+## Protected Header
 
-```json
-{
-   "kty":"EC",
-   "crv":"BLS12381_G1",
-   "use":"prf",
-   "x":"tCgCNuUYQotPEsrljWi-lIRIPpzhqsnJV1NPnE7je6glUb-FJm9IYkuv2hbHw22i",
-   "payloads":[
-      {"claims":["family_name", "given_name"]},
-      {"claims":["email"]},
-      {"claims":["birthdate"]},
-      {"claims":["age"], "cty":"hashchain-commitment"},
-      {"claims":["profile_pic"], "cty":"image/png"}
-   ]
-  }
-```
+Although there are multiple payloads, the protected header still represents the JWP as a whole.
 
-## Protected Header
+It is recommended that payload-specific information is not included in the header and is handled outside of the cryptographic envelope.  This is to minimize any correlatable signals in the metadata, to prevent a verifier from categorizing based on header changes that may varry between multiple JWPs.
```

Comment:
I agree with this, but we should beware of scenarios where time information MUST be disclosed...

proving I have a DL, without proving it's not expired could be a problem... pushing that information to the payload could cause interop issues, similar to `nbf`, `iat`....

Reply:
We should talk about "MUST be disclosed" payloads on the next DIF Crypto JWP call.  It's a great topic.  That said, I don't think this PR should block on solving that.

Reply:
I believe we'll be forming a sibling JWP spec that adds semantics around specific claims like JWT. Presentation exchange is also a good place for these types of rules.

I don't think these payload-specific requirements need to be handled at the core JWP layer other than integrity protecting and disclosing the fixed header.